CMakeLists.txt

```diff
@@ -451,6 +465,25 @@ if(__BUILD_FOR_R)
   endif(MSVC)
 endif(__BUILD_FOR_R)
 
+#-- Google C++ tests
+if(BUILD_CPP_TEST)
+  find_package(GTest CONFIG)
+  if(NOT GTEST_FOUND)
+    message(STATUS "Did not find Google Test in the system root. Fetching Google Test now...")
+    include(FetchContent)
+    FetchContent_Declare(
+      googletest
+      GIT_REPOSITORY https://github.com/google/googletest.git
+      GIT_TAG        release-1.10.0
+    )
+    FetchContent_MakeAvailable(googletest)
+    add_library(GTest::GTest ALIAS gtest)
+  endif()
+  file(GLOB CPP_TEST_SOURCES tests/cpp_test/*.cpp)
+  add_executable(testlightgbm ${CPP_TEST_SOURCES} ${SOURCES})
+  target_link_libraries(testlightgbm PRIVATE GTest::GTest)
```

Comment:
Do new target names mean that we cannot combine other options like `USE_MPI` or `USE_OPENMP` with `BUILD_CPP_TEST`?

Reply:
For now, that's the case. To apply these options, we'll need to create an object library target `objlightgbm` to which we'd apply the set of common options, and then have `lightgbm` and `testlightgbm` depend on `objlightgbm`.

Reply:
OK, got it! Thank a lot for the detailed explanation and possible future workaround!

Reply:
Creating an object library will require a large refactor of the CMake file, so it won't be done in this PR.

Reply:
Yeah, I understand.